plugins/tickets.py

```diff
@@ -0,0 +1,1821 @@
+                        duration = None
+
+                    # Extract comment
+                    comment = content[match.end():]
+                else:
+                    duration = None
+                    comment = content
+
+                # Update the ticket
+                with ticket.batch_update():
+                    ticket.stage = TicketStage.COMMENTED
+                    ticket.duration = duration if (ticket.can_revert and ticket.active) else None
+                    ticket.comment = comment
+                    ticket.modified_by = self.modid
+                    if ticket.status == TicketStatus.NEW:
+                        ticket.status = TicketStatus.IN_EFFECT
+
+                self.last_read_msgid = message.id
+
+                # Notify the moderator, nullify the duration if required
+                if duration:
+                    if not ticket.can_revert:
+                        msg = ("Ticket comment set! "
+                               "Provided duration ignored since this ticket type cannot expire.")
+                        duration = None
+                    elif not ticket.active:
+                        msg = ("Ticket comment set! "
+                               "Provided duration ignored since this ticket is no longer in effect.")
+                        duration = None
+                    else:
+                        msg = "Ticket comment and duration set!"
+                else:
+                    msg = "Ticket comment set!"
+
+                await self.user.send(msg)
+
+                # Publish the ticket, which will also trigger an update of the local ticket embed
+                await self.current_ticket.publish()
+
+                # Schedule ticket expiration, if required
+                if duration:
+                    update_expiry_for(self.current_ticket)
+
+                # Deliver the next ticket
+                await self.deliver()
+            else:
+                self.last_read_msgid = message.id
+
+
+async def reload_mods():
+    """
+    Reload all moderators from data.
+    """
+    global _ticketmods
+
+    logger.debug("Loading ticket moderators.")
+    [mod.unload() for mod in _ticketmods.values()]
+    _ticketmods = {row[0]: TicketMod(row) for row in TicketMod._select_where()}
+    [await mod.load() for mod in _ticketmods.values()]
+    logger.info("Loaded {} ticket moderators.".format(len(_ticketmods)))
+
+
+def get_or_create_mod(modid) -> TicketMod:
+    """
+    Get a single TicketMod by modid, or create it if it doesn't exist.
+    """
+    mod = _ticketmods.get(modid, None)
+    if not mod:
+        mod = TicketMod(TicketMod._insert(modid=modid))
+        _ticketmods[modid] = mod
+    return mod
+
+
+# ------------ Commands ------------
+
+def resolve_ticket(msg, ticketarg=None) -> Ticket:
+    if ticketarg is not None and isinstance(ticketarg, commands.StringArg):
+        maybe_id = int(ticketarg.text)
+        if maybe_id < 2147483647:
```

Comment:
what

Reply:
Hacky way of differentiating between a ticket id and a ticket list message id (snowflake) based on the maximum size of an INT column. Not sure of a better way of doing this, but this *does* clearly differentiate them.

Reply:
I think I'd go with `maybe_id < 2^(10+12)`. The snowflake format uses the leading 42 bits for the timestamp, and a snowflake of 0x3FFFFF would imply a timestamp of 0 (milliseconds). Definitely deserves a comment in the code in either case.

Reply:
That's a cleaner expression of it, thanks!